File: app/src/main/java/com/neo/multichanneltransfer/services/ConnectionService.java
package com.neo.multichanneltransfer.services;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.hardware.usb.UsbManager;
import com.neo.multichanneltransfer.models.ConnectionInfo;
import java.util.HashMap;

public class ConnectionService {
    private Context context;
    private HashMap<String, ConnectionInfo> connections;

    public ConnectionService(Context context) {
        this.context = context;
        this.connections = new HashMap<>();
        initializeConnections();
    }

    private void initializeConnections() {
        connections.put("wifi", new ConnectionInfo("wifi"));
        connections.put("usb", new ConnectionInfo("usb"));
    }

    public void detectConnections() {
        // Check WiFi connection
        ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo wifiInfo = cm.getNetworkInfo(ConnectivityManager.TYPE_WIFI);

        ConnectionInfo wifiConn = connections.get("wifi");
        if (wifiInfo != null && wifiInfo.isConnected()) {
            wifiConn.setActive(true);
            wifiConn.setConnected(true);
            wifiConn.setStatus("connected");
        } else {
            wifiConn.setActive(false);
            wifiConn.setConnected(false);
            wifiConn.setStatus("disconnected");
        }

        // Check USB connection (simplified)
        UsbManager usbManager = (UsbManager) context.getSystemService(Context.USB_SERVICE);
        ConnectionInfo usbConn = connections.get("usb");

        // This is a simplified check - you'd need more complex logic for actual USB detection
        boolean usbConnected = usbManager.getDeviceList().size() > 0;
        usbConn.setActive(usbConnected);
        usbConn.setConnected(usbConnected);
        usbConn.setStatus(usbConnected ? "connected" : "disconnected");
    }

    public HashMap<String, ConnectionInfo> getConnections() {
        return connections;
    }

    public ConnectionInfo getConnection(String type) {
        return connections.get(type);
    }

    public boolean isConnectionAvailable(String type) {
        ConnectionInfo conn = connections.get(type);
        return conn != null && conn.isActive() && conn.isConnected();
    }
}

File: app/src/main/java/com/neo/multichanneltransfer/services/ConnectionService.kt
package com.neo.multichanneltransfer.services

import android.content.Context
import android.hardware.usb.UsbManager
import android.net.ConnectivityManager
import android.net.NetworkCapabilities
import android.os.Build
import com.neo.multichanneltransfer.models.ConnectionInfo
import com.neo.multichanneltransfer.models.ConnectionStatus
import com.neo.multichanneltransfer.models.ConnectionType

class ConnectionService(private val context: Context) {
    private val connections = mutableMapOf<ConnectionType, ConnectionInfo>()

    init {
        connections[ConnectionType.WIFI] = ConnectionInfo(ConnectionType.WIFI)
        connections[ConnectionType.USB] = ConnectionInfo(ConnectionType.USB)
    }

    fun detectConnections() {
        detectWifiConnection()
        detectUsbConnection()
    }

    fun getConnection(type: ConnectionType): ConnectionInfo? = connections[type]

    fun getAllConnections(): Map<ConnectionType, ConnectionInfo> = connections.toMap()

    private fun detectWifiConnection() {
        val wifiConn = connections[ConnectionType.WIFI] ?: return
        val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager

        val isConnected = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            val network = cm.activeNetwork
            val capabilities = network?.let { cm.getNetworkCapabilities(it) }
            capabilities?.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) == true
        } else {
            @Suppress("DEPRECATION")
            val ni = cm.activeNetworkInfo
            ni?.type == ConnectivityManager.TYPE_WIFI && ni.isConnected
        }

        wifiConn.isActive = isConnected
        wifiConn.isConnected = isConnected
        wifiConn.status = if (isConnected) ConnectionStatus.CONNECTED else ConnectionStatus.DISCONNECTED
    }

    private fun detectUsbConnection() {
        val usbConn = connections[ConnectionType.USB] ?: return
        val usbManager = context.getSystemService(Context.USB_SERVICE) as UsbManager
        val deviceList = usbManager.deviceList
        val isConnected = deviceList.isNotEmpty()

        usbConn.isActive = isConnected
        usbConn.isConnected = isConnected
        usbConn.status = if (isConnected) ConnectionStatus.CONNECTED else ConnectionStatus.DISCONNECTED
    }
}
